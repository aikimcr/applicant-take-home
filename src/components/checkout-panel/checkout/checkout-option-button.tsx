import React from 'react';
import { PrizeoutOfferValueOptions } from '../../../slices/offers-slice';

import './checkout.less';

interface CheckoutOptionsButtonsProps {
    giftcard: PrizeoutOfferValueOptions;
    isChecked: boolean;
    onClick: () => void;
}

const CheckoutOptionButton: React.FC<CheckoutOptionsButtonsProps> = ({
    giftcard,
    onClick,
    isChecked,
}): React.ReactElement => {
    const buttonId = `checkout-option-button-${giftcard.checkout_value_id}`;
    const displayAmount = '$' + (giftcard.cost_in_cents / 100).toFixed(2);
    return (
        <div className="grid__item">
            <div className="checkout-radio">
                <input
                    type="radio"
                    name="giftcard-option"
                    value={giftcard.cost_in_cents}
                    id={buttonId}
                    defaultChecked={isChecked}
                />
                <label htmlFor={buttonId} onClick={onClick}>
                    {displayAmount}
                </label>
            </div>
        </div>
    );
};

export default CheckoutOptionButton;
