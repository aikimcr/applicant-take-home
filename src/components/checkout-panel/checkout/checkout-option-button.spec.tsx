import { render, screen } from '../../../testing/test-utils';
import React from 'react';
import '@testing-library/jest-dom';

import { CheckoutOptionButton } from './checkout-option-button';
import { PrizeoutOfferValueOptions } from '../../../slices/offers-slice';

describe('Test CheckoutOptionButton', () => {
    const testCard: PrizeoutOfferValueOptions = {
        checkout_value_id: 'xyzzy',
        cost_in_cents: 1212,
        display_bonus: 33,
        display_monetary_bonus: null,
        value_in_cents: 1612,
    };

    test('Component matches snapshot', () => {
        const clickHandler = jest.fn();

        const { asFragment } = render(
            <CheckoutOptionButton isChecked={false} giftcard={testCard} onClick={clickHandler} />,
        );

        expect(asFragment()).toMatchSnapshot();
    });

    test('Click handler is called', () => {
        const clickHandler = jest.fn();

        render(<CheckoutOptionButton isChecked={false} giftcard={testCard} onClick={clickHandler} />);

        expect(clickHandler).toBeCalledTimes(0);

        const button = screen.queryByText('$12.12');
        expect(button).toBeVisible();
        button.click();

        expect(clickHandler).toBeCalledTimes(1);
    });
});
