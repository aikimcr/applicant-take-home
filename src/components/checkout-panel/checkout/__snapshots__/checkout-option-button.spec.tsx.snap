// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test CheckoutOptionButton Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="grid__item"
  >
    <div
      class="checkout-radio"
    >
      <input
        id="checkout-option-button-xyzzy"
        name="giftcard-option"
        type="radio"
        value="1212"
      />
      <label
        for="checkout-option-button-xyzzy"
      >
        $12.12
      </label>
    </div>
  </div>
</DocumentFragment>
`;
